src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -518,12 +607,16 @@ static NamedArgsAndClosure parseArgs(Object arg, StepDescriptor d) {
             if (singleArgumentOnly) {  // Can fetch the one argument we need
                 DescribableParameter dp = stepModel.getSoleRequiredParameter();
                 String paramName = (dp != null) ? dp.getName() : null;
-                return parseArgs(arg, d.takesImplicitBlockArgument(), paramName, singleArgumentOnly);
+                return parseArgs(arg, d.takesImplicitBlockArgument(), paramName, singleArgumentOnly, new HashSet<>());
             }
         } catch (NoStaplerConstructorException e) {
             // Ignore steps without databound constructors and treat them as normal.
         }
-        return parseArgs(arg,d.takesImplicitBlockArgument(), loadSoleArgumentKey(d), singleArgumentOnly);
+        return parseArgs(arg,d.takesImplicitBlockArgument(), loadSoleArgumentKey(d), singleArgumentOnly, new HashSet<>());
+    }
+
+    static NamedArgsAndClosure parseArgs(Object arg, boolean expectsBlock, String soleArgumentKey, boolean singleRequiredArg) {
+        return parseArgs(arg, expectsBlock, soleArgumentKey, singleRequiredArg, new HashSet<>());
```

Comment:
Is this still used? If not, you can delete it. There shouldn't be any need to keep it to avoid backwards incompatibility or anything since it was a package-private method.

Reply:
This `parseArgs` is used in `invokeDescribable` here: https://github.com/car-roll/workflow-cps-plugin/blob/b14f8cf2ce07227ade95c88d2346ced7b14ad544/src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java#L409

Reply:
Oh ok, maybe clearer to change the call to use the new overload though and delete this one.

Reply:
done in d489f73f2d33ad7eecf0b99bbea2e1291d72a35f